core/common/test/InstantTest.kt

```diff
@@ -574,6 +559,12 @@ class InstantRangeTest {
             assertArithmeticFails("$instant") { instant.plus(Long.MAX_VALUE, DateTimeUnit.YEAR, UTC) }
             assertArithmeticFails("$instant") { instant.plus(Long.MIN_VALUE, DateTimeUnit.YEAR, UTC) }
         }
+        for (instant in smallInstants) {
+            instant.plus(2 * Int.MAX_VALUE.toLong(), DateTimeUnit.DAY, UTC)
+            instant.plus(2 * Int.MIN_VALUE.toLong(), DateTimeUnit.DAY, UTC)
+            instant.plus(2 * Int.MAX_VALUE.toLong(), DateTimeUnit.MONTH, UTC)
+            instant.plus(2 * Int.MIN_VALUE.toLong(), DateTimeUnit.MONTH, UTC)
```

Comment:
Perhaps worth checking if subtracting the same value produces the same instant without clamping in between.